tests/php/ORM/Utf8/Utf8TestHelper.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace SilverStripe\Tests\ORM\Utf8;
+
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Dev\TestOnly;
+use SilverStripe\ORM\DB;
+
+class Utf8TestHelper implements TestOnly
+{
+    use Injectable;
+
+    private string $dbVersion = '';
+
+    public function getUpdatedUtfCharset(string $charset): string
+    {
+        if ($charset !== 'utf8') {
+            return $charset;
+        }
+        return $this->isMySqlGte80() || $this->isMariaDBGte106() ? 'utf8mb3' : 'utf8';
+    }
+
+    public function getUpdatedUtfCollation(string $collation): string
+    {
+        if ($collation === 'utf8_general_ci') {
+            return $this->isMariaDBGte106() ? 'utf8mb3_general_ci' : 'utf8_general_ci';
+        }
+        if ($collation === 'utf8_unicode_520_ci') {
+            return $this->isMariaDBGte106() ? 'utf8mb3_unicode_520_ci' : 'utf8_unicode_520_ci';
+        }
+        return $collation;
+    }
+
+    /**
+     * MySQL has used utf8 as an alias for utf8mb3
+     * Beginning with MySQL 8.0.28, utf8mb3 is used
+     * https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-utf8mb3.html
+     */
+    private function isMySqlGte80(): bool
+    {
+        if (preg_match('#^([0-9]+)\.[0-9]+\.[0-9]+$#', $this->getDBVersion(), $m)) {
+            return (int) $m[1] >= 8;
+        }
+        return false;
+    }
+
+    /**
+     * Until MariaDB 10.5, utf8mb3 was an alias for utf8.
+     * From MariaDB 10.6, utf8 is by default an alias for utf8mb3
+     * https://mariadb.com/kb/en/unicode/
+     */
+    private function isMariaDBGte106(): bool
+    {
+        if (preg_match('#^([0-9]+)\.([0-9]+)\.[0-9]+-mariadb#', $this->getDBVersion(), $m)) {
+            return (int) $m[1] >= 11 || ((int) $m[1] >= 10 && (int) $m[2] >= 6);
+        }
+        return false;
+    }
+
+    private function getDBVersion(): string
+    {
+        if ($this->dbVersion === '') {
+            $this->dbVersion = DB::query('SELECT VERSION();')->first()['VERSION()'];
+            $this->dbVersion = strtolower($this->dbVersion);
```

Comment:
```suggestion
            $this->dbVersion = strtolower(DB::get_conn()->getVersion());
```

Reply:
If this doesn't work for all DB types just say so and don't change it - but at least for mysql it seems to give the same output and is a little cleaner code.

Reply:
Updated

Reply:
```php
$this->dbVersion = DB::query('SELECT VERSION();')->first()['VERSION()'] ?? '';
$this->dbVersion = strtolower($this->dbVersion);
```
It's still the same?

Reply:
No, I added ` ?? '';` so that it won't break on other DB's, it'll just return an empty string.  I changed private $dbVersion from `string` to `?string` where null is "hasn't yet been attempted to be set"

Reply:
Updated to use `DB::get_conn()->getVersion()`